pkgs/pub_formats/lib/pubspec_formats.dart

```diff
@@ -8,4 +8,5 @@
 /// SDK, so when doing breaking changes please roll ASAP to the Dart SDK.
 library;
 
-export 'src/pubspec_lock_syntax.g.dart';
+export 'src/pubspec_syntax.g.dart' hide JsonObjectSyntax, JsonReader;
+export 'src/pubspec_lock_syntax.g.dart' hide JsonObjectSyntax, JsonReader;
```

Comment:
Why are JsonObjectSyntax and JsonReader public classes to begin with? They look like implementation details we only copy in the generated code so that we don't have to depend on another package? Should they just be private?